### DISCUSSION TITLE
Week/Month/Quarter/Year pickers

### DISCUSSION THREAD

**Discussion by jerebtw** [mantinedev/mantine#1054] [Feature requests]
### Usage example, including component/hook API

Add a week/month/quarter/year select

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/week

**Comment by rtivital** [comment]
Well, I guess we can also add MonthPicker, YearPicker and QuarterPicker components as well including range variants. Just curious for what are you using week select?

**Reply by tombohub** [reply]
I would use week select in stock analytics. So people can choose and compare analytics for a week.

**Reply by lucas-santosP** [reply]
Ex: A page that lists payments made per week, I would like to have a filter to limit viewing strictly by weeks and not a free range of two date values.

The week picker is indeed a less frequent use case, but would be useful to have it. And right now is the only "date-related" picker missing in the lib.

**Reply by rtivital** [reply]
https://mantine.dev/dates/calendar/#custom-date-pickers

**Reply by vishnup-ditto** [reply]
![image](https://github.com/mantinedev/mantine/assets/113330036/141a8b80-6e76-4e75-bc99-389a3aa345d5)

We built something like on top of mantine date picker for our Finance platform. Can share some implementation details if this is similar to what you are looking for.

**Reply by tombohub** [reply]
yes, and quarter picker is also useful for financial analytics

**Reply by santisemhan** [reply]
> ![image](https://private-user-images.githubusercontent.com/113330036/254554609-141a8b80-6e76-4e75-bc99-389a3aa345d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA2NzY5NzgsIm5iZiI6MTc0MDY3NjY3OCwicGF0aCI6Ii8xMTMzMzAwMzYvMjU0NTU0NjA5LTE0MWE4YjgwLTZlNzYtNGU3NS1iYzk5LTM4OWEzYWEzNDVkNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyN1QxNzE3NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kZDA3YzEwODk2MDBiZjJmZjgwMTIyYzA5Y2JmNWMzOTNlMGU1ZjBkNzAzODBmY2ZmN2ZiNDY1ZGViNzI2NDcyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.-DpN8S0EY23jxz2WDCuuS6xcKVEtwVLx_biq4WrIxjM)
> 
> We built something like on top of mantine date picker for our Finance platform. Can share some implementation details if this is similar to what you are looking for.

Could you do it with Mantine? Do you have the implementation?

**Comment by jerebtw** [comment]
For a timetable app

**Comment by jerebtw** [comment]
@rtivital could you please add the `Help wanted` label because i don't know much about the date package

**Comment by rtivital** [comment]
Help is not wanted, it's a large feature, I'll work on it on my own when I have time for this

**Comment by jerebtw** [comment]
Ohhh okay

**Comment by LucasLemanowicz** [comment]
A more generic implementation of this could be to allow for "Defined Ranges" that the implementer defines. The interface could look something along the lines of:

```typescript
interface DefinedDateRange {
    label: string;
    dateRange: {
        startDate?: Date,
        endDate?: Date,
    }
}
```

and the `DateRangePicker` would take in a `staticRanges: DefinedDateRange[]` prop, similar to what `react-date-range` does: https://hypeserver.github.io/react-date-range/

This would be incredibly useful for me (if you need any additional signal on the value of this feature).

**Reply by BenSilver1337** [reply]
Thanks!

**Comment by vishnup-ditto** [comment]
Hi, I am looking to override the existing DateRange picker as a variation which allows you to one click select last 7 days, last month, last quarter etc. Assume for the generation of reports or data for a particular period. Haven't been very successful though. Very similar to the hyperserver version linked in jerebtw comments. [hypeserver.github.io/react-date-range](https://hypeserver.github.io/react-date-range/). Has anybody been successful in implementing something like this with Mantine.

**Reply by zhulien-ivanov** [reply]
> Hi, I am looking to override the existing DateRange picker as a variation which allows you to one click select last 7 days, last month, last quarter etc. Assume for the generation of reports or data for a particular period. Haven't been very successful though. Very similar to the hyperserver version linked in jerebtw comments. [hypeserver.github.io/react-date-range](https://hypeserver.github.io/react-date-range/). Has anybody been successful in implementing something like this with Mantine.

Make the `DateRangePicker` controlled and assign the date ranges yourself (on button click). It is pretty straightforward.

**Reply by vishnup-ditto** [reply]
I understand the use of the `value` prop in selecting the ranges I want, that's cool. I wanted to override the UI of datepicker popover, modal to include these button as sidebar or something.

**Reply by vishnup-ditto** [reply]
One way that came to mind is to build my own rangepicker using the calendar base and as such, with my own custom UI incorporated? Just wanted to enquire if there is a easier way to achieve this than that.

**Reply by zhulien-ivanov** [reply]
I solve this by using a button next to the picker that opens a popover with button options.

**Reply by vishnup-ditto** [reply]
Okay, So something like 
``` tsx
  <Popover>
    <Popover.Target>
      <Button>Open DatePicker</Button>
    </Popover.Target>
    <Popover.Dropdown>
      <OtherUIWeWant/>
      <DateRangePicker   />
    </Popover.Dropdown>
  </Popover>
```

**Comment by GuiPretto** [comment]
Hi! I'm using Mantine for the first time this month and I'm really enjoying it, congrats on developing this framework. But I'm here for this discussion, @rtivital you said earlier that you would make this yourself, but its been a long time already, do you still have any plans on doing it? And if not, am I able to do it for you? Currently this component is one of my necessities on the projects I'm working with.

**Reply by rtivital** [reply]
https://rtivital.github.io/mantine-dates-6/

**Reply by GuiPretto** [reply]
Nice, thank you.

**Reply by telmen** [reply]
hi @rtivital, any eta on v6 release?

**Reply by rtivital** [reply]
No, there is not

**Reply by tombohub** [reply]
@rtivital thanks, do you have in plan to make quarter picker?
